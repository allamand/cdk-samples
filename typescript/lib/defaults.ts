export const DEFAULT_CLUSTER_VERSION = '1.17'
export const DEFAULT_CLUSTER_NAME = 'CdkEksCluster'
export const DEFAULT_EXTERNAL_DNS_POLICY = 'sync'
export const DEFAULT_HOSTED_ZONE = 'demo3.allamand.com'
export const DEFAULT_KEY_NAME = 'sallaman'
export const DEFAULT_SSH_SOURCE_IP_RANGE = '0.0.0.0/0'
export const DEFAULT_VPC_IP_RANGE = '10.0.0.0/16'
