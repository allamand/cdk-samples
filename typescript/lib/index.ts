export * from './bastion-linux';
export * from './ecs';
export * from './eks';
export * from './eks-nginx-svc';
export * from './fargate-alb-svc';
export * from './fargate-cicd';
export * from './fargate-event-targets';
export * from './lambda-filesystem';
export * from './autoscaling-group';
export * from './vpc';
export * from './vpc-client-vpn';
export * from  './transcribe';
export * from  './apig-custom-domain';
export * from  './apisix';
export * from  './serverless-app';
export * from  './serverless-rest-api';
