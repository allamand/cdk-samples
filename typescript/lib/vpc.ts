import cdk = require('@aws-cdk/core');
import ec2 = require('@aws-cdk/aws-ec2');
import {DEFAULT_CLUSTER_VERSION, DEFAULT_VPC_IP_RANGE} from "./defaults";
import {Stack} from "@aws-cdk/core";

export class VpcProvider extends cdk.Stack {

    public static getOrCreate(scope: cdk.Construct) {



        const stack = cdk.Stack.of(scope)
        const value = stack.node.tryGetContext('use_vpc_id')

        const vpcIpRange = stack.node.tryGetContext('vpc_ip_cidr') ?? DEFAULT_VPC_IP_RANGE
        const vpc = stack.node.tryGetContext('use_default_vpc') === '1' ?
        ec2.Vpc.fromLookup(stack, 'Vpc', { isDefault: true }) :
            stack.node.tryGetContext('use_vpc_id') ?
            ec2.Vpc.fromLookup(stack, 'Vpc', { vpcId: stack.node.tryGetContext('use_vpc_id') }) :
                new ec2.Vpc(stack, 'Vpc', {
                    maxAzs: 4,
                    natGateways: 1,
                    cidr: vpcIpRange,
                    subnetConfiguration: [
                        {
                            cidrMask: 20,
                            name: 'Public',
                            subnetType: ec2.SubnetType.PUBLIC,
                        },
                        {
                            cidrMask: 20,
                            name: 'Private',
                            subnetType: ec2.SubnetType.PRIVATE,
                        },
                    ],
                });

    return vpc
    }
}

export class VpcStack extends cdk.Stack {
    constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
        super(scope, id, props);

        const vpc = VpcProvider.getOrCreate(this)

        new cdk.CfnOutput(this, 'VpcId', { value: vpc.vpcId })
            }
}

// const vpc = VpcProvider.getOrCreate(this)


// function getOrCreateVpc(stack: cdk.Stack): ec2.IVpc {
//     // use an existing vpc or create a new one
//     const vpc = stack.node.tryGetContext('use_default_vpc') === '1' ?
//         ec2.Vpc.fromLookup(stack, 'Vpc', { isDefault: true }) :
//         stack.node.tryGetContext('use_vpc_id') ?
//             ec2.Vpc.fromLookup(stack, 'Vpc', { vpcId: stack.node.tryGetContext('use_vpc_id') }) :
//             new ec2.Vpc(stack, 'Vpc', { maxAzs: 3, natGateways: 1 });

//     return vpc
// }
