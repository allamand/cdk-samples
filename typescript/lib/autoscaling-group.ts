import * as cdk from '@aws-cdk/core';
import * as ec2 from '@aws-cdk/aws-ec2';
import * as autoscaling from '@aws-cdk/aws-autoscaling';
import { VpcProvider } from './vpc';

export class AutoscalingGroupStack extends cdk.Stack {
  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    const vpc = VpcProvider.getOrCreate(this);

    // create ASG with explicit subnet IDs
    new autoscaling.AutoScalingGroup(this, 'ASG', {
      instanceType: new ec2.InstanceType('t3.large'),
      machineImage: new ec2.AmazonLinuxImage(),
      vpc,
      vpcSubnets: {
        subnets: [
          ec2.Subnet.fromSubnetAttributes(this, 'subnetA', {
            subnetId: 'subnet-aaaa',
            availabilityZone: 'ap-northeast-1a',
          }),
          ec2.Subnet.fromSubnetAttributes(this, 'subnetB', {
            subnetId: 'subnet-bbbb',
            availabilityZone: 'ap-northeast-1b',
          }),
          ec2.Subnet.fromSubnetAttributes(this, 'subnetC', {
            subnetId: 'subnet-cccc',
            availabilityZone: 'ap-northeast-1c',
          }),
        ]
      }
    })
  }
}
