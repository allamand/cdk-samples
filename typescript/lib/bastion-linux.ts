import * as ec2 from '@aws-cdk/aws-ec2';
import * as cdk from '@aws-cdk/core';
import { VpcProvider } from './vpc';

export class BastionHost extends cdk.Stack {
  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    const vpc = VpcProvider.getOrCreate(this)
    const bas = new ec2.BastionHostLinux(this, 'Bastion', {
      vpc,
      subnetSelection: { subnetType: ec2.SubnetType.PUBLIC },
      instanceType: new ec2.InstanceType('t3.large'),
    })

    const homeIp = this.node.tryGetContext('HOME_IP') || undefined
    if(homeIp) {
      console.debug("using homeIp "+ homeIp);
      bas.allowSshAccessFrom(ec2.Peer.ipv4(`${homeIp}/32`))
    }

    new cdk.CfnOutput(this, 'instanceId', { value: bas.instanceId })
    new cdk.CfnOutput(this, 'az', { value: bas.instanceAvailabilityZone })
    new cdk.CfnOutput(this, 'publicIp', { value: bas.instancePublicIp })
    new cdk.CfnOutput(this, 'runCommand', { value: 
      `aws ec2-instance-connect send-ssh-public-key --instance-id ${bas.instanceId} --instance-os-user ec2-user 
      --ssh-public-key file://{PUBLIC_KEY_PATH} --availability-zone ${bas.instanceAvailabilityZone}` })
  }
}
