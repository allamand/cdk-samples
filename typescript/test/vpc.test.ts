import {App} from "@aws-cdk/core";
import * as vpc from "../lib/vpc";
import {SynthUtils} from "@aws-cdk/assert";


test('Test vpc creation', () => {

    const app = new App();
    const stack = new vpc.VpcProvider(app, 'VPC');
    const myVpc = vpc.VpcProvider.getOrCreate(stack);
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
});



test('Test use use_vpc_id', () => {
    const app = new App(  )

    const stack = new vpc.VpcProvider(app, 'VPC',  {
        env: {
            account: process.env.CDK_DEFAULT_ACCOUNT,
            region: process.env.CDK_DEFAULT_REGION
            },
    });
    stack.node.setContext("use_vpc_id", 'vpc-xxxxxxxx');
    const myVpc = vpc.VpcProvider.getOrCreate(stack);
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
});

