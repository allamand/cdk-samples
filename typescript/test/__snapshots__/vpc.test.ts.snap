// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test use use_vpc_id 1`] = `Object {}`;

exports[`Test vpc creation 1`] = `
Object {
  "Resources": Object {
    "Vpc8378EB38": Object {
      "Properties": Object {
        "CidrBlock": "10.0.0.0/16",
        "EnableDnsHostnames": true,
        "EnableDnsSupport": true,
        "InstanceTenancy": "default",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VPC/Vpc",
          },
        ],
      },
      "Type": "AWS::EC2::VPC",
    },
    "VpcIGWD7BA715C": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VPC/Vpc",
          },
        ],
      },
      "Type": "AWS::EC2::InternetGateway",
    },
    "VpcPrivateSubnet1DefaultRouteBE02A9ED": Object {
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "NatGatewayId": Object {
          "Ref": "VpcPublicSubnet1NATGateway4D7517AA",
        },
        "RouteTableId": Object {
          "Ref": "VpcPrivateSubnet1RouteTableB2C5B500",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "VpcPrivateSubnet1RouteTableAssociation70C59FA6": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "VpcPrivateSubnet1RouteTableB2C5B500",
        },
        "SubnetId": Object {
          "Ref": "VpcPrivateSubnet1Subnet536B997A",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "VpcPrivateSubnet1RouteTableB2C5B500": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VPC/Vpc/PrivateSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "VpcPrivateSubnet1Subnet536B997A": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            0,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.32.0/20",
        "MapPublicIpOnLaunch": false,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Private",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Private",
          },
          Object {
            "Key": "Name",
            "Value": "VPC/Vpc/PrivateSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "VpcPrivateSubnet2DefaultRoute060D2087": Object {
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "NatGatewayId": Object {
          "Ref": "VpcPublicSubnet1NATGateway4D7517AA",
        },
        "RouteTableId": Object {
          "Ref": "VpcPrivateSubnet2RouteTableA678073B",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "VpcPrivateSubnet2RouteTableA678073B": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VPC/Vpc/PrivateSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "VpcPrivateSubnet2RouteTableAssociationA89CAD56": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "VpcPrivateSubnet2RouteTableA678073B",
        },
        "SubnetId": Object {
          "Ref": "VpcPrivateSubnet2Subnet3788AAA1",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "VpcPrivateSubnet2Subnet3788AAA1": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            1,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.48.0/20",
        "MapPublicIpOnLaunch": false,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Private",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Private",
          },
          Object {
            "Key": "Name",
            "Value": "VPC/Vpc/PrivateSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "VpcPublicSubnet1DefaultRoute3DA9E72A": Object {
      "DependsOn": Array [
        "VpcVPCGWBF912B6E",
      ],
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": Object {
          "Ref": "VpcIGWD7BA715C",
        },
        "RouteTableId": Object {
          "Ref": "VpcPublicSubnet1RouteTable6C95E38E",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "VpcPublicSubnet1EIPD7E02669": Object {
      "Properties": Object {
        "Domain": "vpc",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VPC/Vpc/PublicSubnet1",
          },
        ],
      },
      "Type": "AWS::EC2::EIP",
    },
    "VpcPublicSubnet1NATGateway4D7517AA": Object {
      "Properties": Object {
        "AllocationId": Object {
          "Fn::GetAtt": Array [
            "VpcPublicSubnet1EIPD7E02669",
            "AllocationId",
          ],
        },
        "SubnetId": Object {
          "Ref": "VpcPublicSubnet1Subnet5C2D37C4",
        },
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VPC/Vpc/PublicSubnet1",
          },
        ],
      },
      "Type": "AWS::EC2::NatGateway",
    },
    "VpcPublicSubnet1RouteTable6C95E38E": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VPC/Vpc/PublicSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "VpcPublicSubnet1RouteTableAssociation97140677": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "VpcPublicSubnet1RouteTable6C95E38E",
        },
        "SubnetId": Object {
          "Ref": "VpcPublicSubnet1Subnet5C2D37C4",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "VpcPublicSubnet1Subnet5C2D37C4": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            0,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.0.0/20",
        "MapPublicIpOnLaunch": true,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Public",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Public",
          },
          Object {
            "Key": "Name",
            "Value": "VPC/Vpc/PublicSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "VpcPublicSubnet2DefaultRoute97F91067": Object {
      "DependsOn": Array [
        "VpcVPCGWBF912B6E",
      ],
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": Object {
          "Ref": "VpcIGWD7BA715C",
        },
        "RouteTableId": Object {
          "Ref": "VpcPublicSubnet2RouteTable94F7E489",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "VpcPublicSubnet2RouteTable94F7E489": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VPC/Vpc/PublicSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "VpcPublicSubnet2RouteTableAssociationDD5762D8": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "VpcPublicSubnet2RouteTable94F7E489",
        },
        "SubnetId": Object {
          "Ref": "VpcPublicSubnet2Subnet691E08A3",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "VpcPublicSubnet2Subnet691E08A3": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            1,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.16.0/20",
        "MapPublicIpOnLaunch": true,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Public",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Public",
          },
          Object {
            "Key": "Name",
            "Value": "VPC/Vpc/PublicSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "VpcVPCGWBF912B6E": Object {
      "Properties": Object {
        "InternetGatewayId": Object {
          "Ref": "VpcIGWD7BA715C",
        },
        "VpcId": Object {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::VPCGatewayAttachment",
    },
  },
}
`;
